File: APIQuanLyCuaHang/APIQuanLyCuaHang/ViewModels/DataChart/EarningData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace APIQuanLyCuaHang.ViewModels.DataChart
{
    public class EarningData
    {
        public string? Name { get; set; }
        public List<decimal>? Data { get; set; }
        public List<string>? Categories { get; set; }
    }
}

File: QuanLyCuaHangUXUI/QuanLyCuaHangProject/src/models/dto/dataCharts/EarningData.js
class EarningData {
  constructor(name = '', data = [], categories = []) {
    this.name = name
    this.data = data
    this.categories = categories
  }

  static fromJson(json) {
    return new EarningData(json.name || '', json.data || [], json.categories || [])
  }

  toJson() {
    return {
      name: this.name,
      data: this.data,
      categories: this.categories,
    }
  }
}

export default EarningData
